Sources/SwiftDependencyGraphs/DependencyGraph.swift

```diff
@@ -105,6 +108,23 @@ public struct DependencyGraph<V> where V: Hashable, V: Identifiable {
     return edges[vertex.id]
   }
 
+  /// A neighbour of a vertex is connected to it by an edge. It computes the neighbours
+  /// for incoming and outgoing edges.
+  /// - Parameters:
+  ///   - vertex: The vertex for which the neighbours are computed.
+  /// - Returns: The neighbours of the vertex if the vertex is in the graph and `nil` otherwise.
+  public func neighbours(of vertex: V) -> OrderedSet<V>? {
+    guard let forwards = neighbours(of: vertex, in: .forwards) else {
+      return neighbours(of: vertex, in: .backwards)
+    }
+
+    guard let backwards = neighbours(of: vertex, in: .backwards) else {
+      return forwards
+    }
+
+    return forwards.union(backwards)
```

Comment:
I found the 3 exit points for different conditions interesting; but maybe this function is equivalent to providing fallbacks and a union?

```suggestion
return (neighbours(of: vertex, in: .backwards) ?? []).union(neighbours(of: vertex, in: .backwards) ?? [])
```

Reply:
It is not equivalent because it does not return `nil` if `vertex` has no neighbours in `.backwards` direction, for example. I want to make a distinction there because it makes a difference if a vertex just has no neighbours (empty set) or is not in the dictionary (`nil`).

Reply:
Ah, I see. Didn't think of `let backwards = nil`!